webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx

```diff
@@ -148,15 +149,27 @@ function ResultDetails() {
     },
   );
 
+  // !NOTE: Workaround to display the date in the correct format, to be replaced by a proper solution.
+  const dateTimeFromIndex = useMemo(() => {
```

Comment:
It's better to not mix it with the matrix promise and keep clear separation of concerns. For the `useMemo`, the `index` array in `matrixRes.data` is large, making date formatting expensive. If we trigger a filter for example it will be re-calculated.
don't you think we should keep it?

Reply:
Because it's workaround before the backend fix it, I think it's correct to place it in the usePromise.
As you want.